lightwood/encoders/time_series/helpers/rnn_helpers.py

```diff
@@ -2,58 +2,84 @@
 
 import torch
 import torch.nn as nn
-import torch.nn.functional as F
 
-import numpy as np
 
-
-def tensor_from_series(series, device):
-    if type(series) != type([]):
-        series = str(series).split(' ')
+def tensor_from_series(series, device, n_dims=None, pad_value=0.0, max_len=64):
+    """
+    :param series: list of lists, corresponds to time series: [[x1_1, ..., x1_n], [x2_1, ..., x2_n], ...]
+                   the series is zero-padded on each axis so that all dimensions have equal length
+    :param device: computing device that PyTorch backend uses
+    :param n_dims: if not None, will zero-pad dimensions until series_dimensions = n_dims
+    :param pad_value: value to pad each dimension in the time series, if needed
+    :param max_len: length to pad or truncate each time_series
+    :return: series as a tensor ready for model consumption, shape (1, ts_length, n_dims)
+    """
+    if isinstance(series[0], torch.Tensor):
+        return series[0]
```

Comment:
I feel like this is a decision chain that should be inlined where it happens, unless there's a situation where we call this method there we are unsure if the internal elements are tesnors or lists.